flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -1924,7 +1924,16 @@ else if (typeHierarchy.size() <= 1) {
 				pojoFields.add(new PojoField(field, ti));
 			} catch (InvalidTypesException e) {
 				Class<?> genericClass = Object.class;
-				if(isClassType(fieldType)) {
+				// use information about bounds if available
+				if (fieldType instanceof TypeVariable) {
+					TypeVariable<?> typeVar = (TypeVariable<?>) fieldType;
+					// we take the first bound (usually the class), everything is better than Object
+					if (typeVar.getBounds().length > 0 && isClassType(typeVar.getBounds()[0])) {
+						genericClass = typeToClass(typeVar.getBounds()[0]);
```

Comment:
_getBounds()_ can throw runtime exceptions, should we not fall back to Object if it happens ?